provider/src/main/java/com/synopsys/integration/alert/provider/blackduck/task/accumulator/BlackDuckAccumulator.java

```diff
@@ -54,80 +59,89 @@ public BlackDuckAccumulator(
         ProviderTaskPropertiesAccessor providerTaskPropertiesAccessor,
         ProviderProperties providerProperties,
         BlackDuckValidator blackDuckValidator,
-        EventManager eventManager
+        EventManager eventManager,
+        BlackDuckNotificationRetrieverFactory notificationRetrieverFactory
     ) {
         super(blackDuckProviderKey, taskScheduler, providerProperties);
         this.blackDuckProviderKey = blackDuckProviderKey;
         this.notificationAccessor = notificationAccessor;
         this.blackDuckValidator = blackDuckValidator;
         this.eventManager = eventManager;
+        this.notificationRetrieverFactory = notificationRetrieverFactory;
         this.searchDateManager = new BlackDuckAccumulatorSearchDateManager(providerTaskPropertiesAccessor, providerProperties.getConfigId(), getTaskName());
     }
 
     @Override
-    public void runProviderTask() {
-        if (blackDuckValidator.validate(getProviderProperties())) {
-            accumulate();
-        }
+    public String scheduleCronExpression() {
+        return ScheduledTask.EVERY_MINUTE_CRON_EXPRESSION;
     }
 
     @Override
-    public String scheduleCronExpression() {
-        return ScheduledTask.EVERY_MINUTE_CRON_EXPRESSION;
+    protected void runProviderTask() {
+        if (blackDuckValidator.validate(getProviderProperties())) {
+            accumulateNotifications();
+        }
     }
 
     @Override
     protected BlackDuckProperties getProviderProperties() {
         return (BlackDuckProperties) super.getProviderProperties();
     }
 
-    public void accumulate() {
-        DateRange dateRange = searchDateManager.retrieveNextSearchDateRange();
-        retrieveAndStoreNotifications(dateRange)
-            .map(latestNotification -> latestNotification.plusNanos(1000000))
-            .ifPresent(searchDateManager::saveNextSearchStart);
+    private void accumulateNotifications() {
+        Optional<BlackDuckNotificationRetriever> optionalNotificationRetriever = notificationRetrieverFactory.createBlackDuckNotificationRetriever(getProviderProperties());
+        if (optionalNotificationRetriever.isPresent()) {
+            DateRange dateRange = searchDateManager.retrieveNextSearchDateRange();
+            logger.info("Accumulating notifications between {} and {} ", DateUtils.formatDate(dateRange.getStart(), RestConstants.JSON_DATE_FORMAT), DateUtils.formatDate(dateRange.getEnd(), RestConstants.JSON_DATE_FORMAT));
```

Comment:
I've seen the DateUtils.formatDate with JSON_DATE_FORMAT used in many places. We should add a formatAsJsonDate method or something similar that doesn't require this constant passed.

Reply:
Creating a separate PR for this.